### PR TITLE
Add usage and some info to README.md

### PR DESCRIPTION
**Pull request by agrubi**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,66 @@
...
+            rel: ['item'],
+            links: [{ rel: ['self'], href: 'http://api.example.com/entity' }],
+          }],
+          actions: [{ name: 'action', href: 'http://api.example.com/action' }],
+        },
+        contentType: 'application/vnd.siren+json',
+      };
+    }
+    return {};
+  }
```
Comment (nathanhoel) [inline README.md:56]: I don't think we need any of this extra context for the readme.
Maybe we could have example of how to setup the client - but the first thing should just be what is contained on lines 59-63

Then maybe followed by setup.
People generally want to see the main use case up front then details after

File: README.md [hunk 2]
```diff
@@ -1,2 +1,24 @@
...
+
+Install from NPM:
+```shell
+npm install fluent-siren-client
+```
+
+## Usage
+
+```javascript
+const client = new SirenClient(requestFn);
```
Comment (nathanhoel) [inline README.md:19]: we should specify what the signature of `requestFn()`
maybe right beside this in a comment?
```
const client = new SirenClient(requestFn); // async requestFn(uri, method, fieldValues)
```

or something like that
I had mentioned before that a section below `Usage` for setup would be appropriate.
You could show maybe a really simple way to setup requestFn with just request-promise, and maybe suggest this is where you could do authorization steps if you have any.

Comment (agrubi) [reply README.md:19]: `As of Feb 11th 2020, request is fully deprecated`. Should we look into different library for implementing `requestFn`, or is it good enough to include an example using the deprecated request-promise?

Comment (nathanhoel) [reply README.md:19]: lets just use request promise in the example for now - people can substitute their own lib, it's just an example.
Request is still in maintenance mode.

**Pull request merged**